server/src/test/java/com/objectcomputing/checkins/services/memberskills/MemberSkillControllerTest.java

```diff
@@ -0,0 +1,223 @@
+                () -> client.toBlocking().exchange(request, Map.class));
+
+        JsonNode body = responseException.getResponse().getBody(JsonNode.class).orElse(null);
+        JsonNode errors = Objects.requireNonNull(body).get("_embedded").get("errors");
+        JsonNode href = Objects.requireNonNull(body).get("_links").get("self").get("href");
+        List<String> errorList = List.of(errors.get(0).get("message").asText(), errors.get(1).get("message").asText())
+                .stream().sorted().collect(Collectors.toList());
+        assertEquals("memberSkill.memberid: must not be null", errorList.get(0));
+        assertEquals("memberSkill.skillid: must not be null", errorList.get(1));
+        assertEquals(request.getPath(), href.asText());
+        assertEquals(HttpStatus.BAD_REQUEST, responseException.getStatus());
+
+        verify(memberSkillServices, never()).save(any(MemberSkill.class));
+    }
+
+    @Test
+    void testCreateANullMemberSkill() {
+        MemberSkill m = new MemberSkill(UUID.randomUUID(), UUID.randomUUID());
+        when(memberSkillServices.save(any(MemberSkill.class))).thenReturn(m);
+
+        final HttpRequest<String> request = HttpRequest.POST("", "");
+        HttpClientResponseException responseException = assertThrows(HttpClientResponseException.class,
+                () -> client.toBlocking().exchange(request, Map.class));
+
+        JsonNode body = responseException.getResponse().getBody(JsonNode.class).orElse(null);
+        JsonNode errors = Objects.requireNonNull(body).get("message");
+        JsonNode href = Objects.requireNonNull(body).get("_links").get("self").get("href");
+        assertEquals("Required Body [memberSkill] not specified", errors.asText());
+        assertEquals(request.getPath(), href.asText());
+        assertEquals(HttpStatus.BAD_REQUEST, responseException.getStatus());
+
+        verify(memberSkillServices, never()).save(any(MemberSkill.class));
+    }
+
+    @Test
+    void deleteMemberSkill_as_admin() {
+        UUID uuid = UUID.randomUUID();
+
+        doAnswer(an -> {
+            assertEquals(uuid, an.getArgument(0));
+            return null;
+        }).when(memberSkillServices).delete(any(UUID.class));
+
+        final HttpRequest<Object> request = HttpRequest.DELETE(uuid.toString()).basicAuth("ADMIN", "ADMIN");
+        final HttpResponse<String> response = client.toBlocking().exchange(request, String.class);
+
+        assertEquals(HttpStatus.OK, response.getStatus());
+
+        verify(memberSkillServices, times(1)).delete(any(UUID.class));
+    }
+    @Test
+    void deleteMemberSkill_not_as_admin() {
+        UUID uuid = UUID.randomUUID();
+
+        doAnswer(an -> {
+            assertEquals(uuid, an.getArgument(0));
+            return null;
+        }).when(memberSkillServices).delete(any(UUID.class));
+
+        final HttpRequest<Object> request = HttpRequest.DELETE(uuid.toString()).basicAuth("MEMBER", "MEMBER");
```

Comment:
Making the controller `@Secured(SecurityRule.IS_AUTHENTICATED)` like I suggested will mean adding `.basicAuth()` calls to all of your requests.

Reply:
👍